drafts/25-xxx-specifications.txt

```diff
@@ -296,25 +296,31 @@ di07. The non-directive directive
 7 Predefined macros
 ===================
 
-pm01. Any macro name predefined by the implementation shall begin with a
+pm00. Macro names beginning with a leading underscore followed by an uppercase
+      letter or a second underscore are reserved to the processor. Reserved
+      names shall not be the subject of a #define or #undef preprocessing
+      directive within a program unit.
+      [Note: Any valid macro name matching the regular expression `/^_[A-Z][A-Za-z0-9_]*/`
+      is considered reserved.]
```

Comment:
The updated regex is incorrect because it fails to match names like `__reserved` that are reserved by the prose.
(I blame @gklimowicz ;-) )

```suggestion
      [Note: Any valid macro name matching the regular expression `/^_[A-Z_][A-Za-z0-9_]*/`
      is considered reserved.]
```

FWIW I'm ambivalent about the leading `^`, as I expect that will be understood. I'm not even sure this "note" is really necessary/beneficial, because it just duplicates what the rule prose already specifies.

Reply:
Oops, you're right again.

Happy to remove the note, though IIUC notes are mostly just there to help the implementers but aren't part of the standard? It might just be that I'm weird and find it easier to parse a regex than a paragraph...